src/main/java/com/gluonhq/substrate/util/FileDeps.java

```diff
@@ -319,4 +387,67 @@ private void downloadJavaFXZip(String osarch, Path substratePath ) throws IOExce
+     */
+    private void downloadAndroidSdkZip() throws IOException {
+        Path sdk = configuration.getAndroidSdkPath();
+        String hostOs = configuration.getHostTriplet().getOs();
+        String androidSdkUrl = Strings.substitute(ANDROID_SDK_URL, Map.of("host", hostOs));
+        Logger.logDebug("Downloading Android SDK...");
+        FileOps.downloadAndUnzip(androidSdkUrl, sdk.getParent(), "android-sdk.zip", sdk.getFileName().toString(), "");
+        Logger.logDebug("Done downloading Android SDK");
+    }
+    /**
+     * Downloads libraries needed for Android SDK's sdkmanager 
+     * @throws IOException in case anything goes wrong.
+     */
+    private void downloadAdditionalAndroidLibs() throws IOException {
+        Path sdk = configuration.getAndroidSdkPath();
+        Path libsLocation = sdk.resolve("tools").resolve("lib").resolve("java11");
+
+        Files.createDirectories(libsLocation);
+        Logger.logDebug("Downloading additional libs ...");
+        for (String url : ANDROID_DEPS) {
+            URL link = new URL(url);
+            String filename = url.substring(url.lastIndexOf('/')+1, url.length());
+            FileOps.downloadFile(link, libsLocation.resolve(filename));
+        }
+        Logger.logDebug("Done downloading additional libs");
+    }
+
+    /**
+     * Runs Android SDK's sdkmanager with specified arguments
+     * @param args array of arguments to be passed to process
+     * @throws IOException in case anything goes wrong.
+     * @throws InterruptedException in case anything goes wrong.
+     */
+    private void androidSdkManager(String[] args) throws IOException, InterruptedException {
+        Path sdk = configuration.getAndroidSdkPath();
+        Path tools = sdk.resolve("tools");
+        Path libs = tools.resolve("lib");
+        Path additionalLibs = libs.resolve("java11");
+
+        ProcessRunner sdkmanager = new ProcessRunner(Paths.get(configuration.getGraalPath().toString(), "bin", "java").toString(), "-Dcom.android.sdklib.toolsdir=" + tools, "-classpath",
+                libs + "/*:" + additionalLibs + "/*", "com.android.sdklib.tool.sdkmanager.SdkManagerCli");
+        sdkmanager.addArgs(args);
+        sdkmanager.setInteractive(true); // Needed to accept EULA and show download progress
+
+        Logger.logDebug("Running sdkmanager with: " + sdkmanager.getCmd());
+        Logger.logDebug("You might be prompted to accept EULA.");
+        sdkmanager.runProcess("sdkmanager");
+    }
+
+    /**
+     * Downloads Android NDK and build tools
+     * @throws IOException in case anything goes wrong.
+     * @throws InterruptedException in case anything goes wrong.
+     */
+    private void fetchFromSdkManager() throws IOException, InterruptedException {
+        Logger.logDebug("Downloading Android toolchain...");
+        String[] args = {"platforms;android-27", "build-tools;27.0.3", "platform-tools", "extras;android;m2repository", "extras;google;m2repository", "ndk-bundle"};
```

Comment:
Missed this before: Can you move this to a constant on top of the file? We shouldn't have these hardcoded versions hidden in a method.